### ISSUE TITLE
android build failed

### ISSUE THREAD

**Issue by clock999** [radareorg/radare2#18868]
Hello,

I am trying to build the android version for radare2. But I failed to build it with below errors. Do you know how to debug the build error and fix that? Thanks a lot!

aarch64-linux-android-ar: creating libr.a
.libr2/libr_asm.a/amd29k.o: In function `amd29k_instr_decode':
/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:431: multiple definition of `amd29k_instr_decode'
.libr2/libr_anal.a/amd29k.o:/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:431: first defined here
.libr2/libr_asm.a/amd29k.o: In function `amd29k_instr_is_ret':
/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:469: multiple definition of `amd29k_instr_is_ret'
.libr2/libr_anal.a/amd29k.o:/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:469: first defined here
.libr2/libr_asm.a/amd29k.o: In function `amd29k_instr_jump':
/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:477: multiple definition of `amd29k_instr_jump'
.libr2/libr_anal.a/amd29k.o:/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:477: first defined here
.libr2/libr_asm.a/amd29k.o: In function `amd29k_instr_print':
/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:498: multiple definition of `amd29k_instr_print'
.libr2/libr_anal.a/amd29k.o:/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:498: first defined here
.libr2/libr_asm.a/amd29k.o: In function `decode_none':
/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:203: multiple definition of `amd29k_instructions'
.libr2/libr_anal.a/amd29k.o:/home/wy/sw/radare2/radare2/libr/asm/p/../arch/amd29k/amd29k.c:203: first defined here
.libr2/libr_asm.a/avr_disasm.o: In function `disassembleInstruction':

clang70: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libr.so] Error 1
make[1]: *** [all] Error 2
make: *** [all] Error 2

**Comment by radare** [comment]
you probably dont want to build those plugins. so you may do:

cp dist/plugins-cfg/plugins.android.cfg plugins.cfg
./configure-plugins

**Comment by clock999** [comment]
Hello,
Thanks for your help! Now I can proceed to build. But I got a new error, and I copied the error message.
I searched the android ndk, but I can't find the bin file which is called ndk-gcc.
Do you think I am using a wrong version of android ndk? 

CC cdb.c
CC buffer.c
make[5]: ndk-gcc: Command not found
make[5]: ndk-gcc: Command not found
make[5]: *** [Makefile:86: cdb.o] Error 127
make[5]: *** Waiting for unfinished jobs....
make[5]: *** [Makefile:86: buffer.o] Error 127
make[4]: *** [Makefile:34: sdb_objs] Error 2
make[3]: *** [Makefile:144: sdb-target] Error 2
make[2]: *** [Makefile:117: sdb/src/libsdb.a] Error 2
make[1]: *** [Makefile:120: sdbs] Error 2
make: *** [Makefile:51: all] Error 2

**Comment by radare** [comment]
This progrm is available in path if you run the android toolchain or use the right scripts to build for android. See sys/android*.sh

**Comment by clock999** [comment]
Hello,
The version of the ndk that I used is 22.1.7171670, which should be a current one. 
I tried to run the below command but failed. The command is copied from the  android-shell.sh.

make-standalone-toolchain.sh --arch=arm64 --install-dir=/tmp/ndk/ --force --api=28

The error is as below.
HOST_OS=linux
HOST_EXE=
HOST_ARCH=x86_64
HOST_TAG=linux-x86_64
HOST_NUM_CPUS=8
BUILD_NUM_CPUS=16
ls: cannot access '/home/wy/Android/Sdk/ndk/22.1.7171670/../prebuilts/ndk/current/platforms': No such file or directory
ERROR: Unknown option '--api'. See --help for usage.

So I tried the below one without --api.

make-standalone-toolchain.sh --arch=arm64 --install-dir=/tmp/ndk/ --force

The standalone ndk files are created under /tmp/ndk/ successfully. But I still can't find the binary file named ndk-gcc.
I am not sure if this is related with below message which is printed when running make-standalone-toolchain.sh.

ls: cannot access '/home/wy/Android/Sdk/ndk/22.1.7171670/../prebuilts/ndk/current/platforms': No such file or directory

Or I am using a unsuitable version of ndk?
I think if the ndk-gcc will be generated into the standalone ndk, it should be one of the original ndk files which is just copied to the install path of standalone ndk.
The r2 script running error is just caused by ndk-gcc. But I really can't find the ndk-gcc file anywhere.
